rfc-139-enable-http3.md

```diff
@@ -0,0 +1,135 @@
+
+Source: ["Head-of-Line Blocking in QUIC and HTTP/3: The Details"](https://calendar.perfplanet.com/2020/head-of-line-blocking-in-quic-and-http-3-the-details/) by [Robin Marx](https://twitter.com/programmingart).
+
+Unfortunately this didn't solve all the HOL blocking issues. Since HTTP/2 is built upon TCP, and TCP is a protocol that offers the reliable transmission of packets. TCPs job is to make sure that the packets leaving one device arrive at another device in the exact same order and without any missing packets. When packet loss occurs, this is when we again see HOL blocking. When a packet of data is lost over the network, this then holds up the whole (single) TCP connection. Packets behind the missing packet can't be processed until retransmission occurs.
+
+This is where we are currently at with GOV.UK. The vast majority of users on strong, stable connection will have seen improved performance from enabling HTTP/2. Unfortunately, some users may have seen performance drop if they use a connection with high packet loss. This is due to the HOL blocking that now exists at the network layer. This is a primary issue HTTP/3 sets out to solve.
+
+### HTTP/3
+To solve the HOL blocking issue at the network layer, HTTP/3 uses a completely new transport layer protocol called QUIC. QUIC initially stood for "Quick UDP Internet Connections", but the IETF now don't use it as an acronym. It is simply the name of the protocol. QUIC is built upon User Datagram Protocol (UDP). Since QUIC uses UDP, it isn't bound by the restrictions that TCP imposes. So it has now been designed so that the network layer now understands the the concept of independent streams, which was restricted to the application layer under HTTP/2:
+
+![Difference between TCP and QUIC when downloading assets.](rfc-139/QUIC-illustration-final.jpg)
+
+In the illustration above, if packet 2 is lost (using HTTP/2), packets 3-8 are blocked until retransmission occurs. Using QUIC if packet 2 is lost only packets 1 and 3 are blocked since they exist in the same stream. The other streams can progress as normal since they are fully independent. QUIC doesn't require in-order packet transmission across the whole connection, but it does still retain ordering within each resource stream.
+
+## The effect of packet loss
+
+Using WebPageTest we can roughly simulate the effect that packet loss has on performance for our users. For testing I picking a fairly content heavy page like the [past Prime Ministers](https://www.gov.uk/government/history/past-prime-ministers) page and a low spec Moto G4 Android device. For the connection speed I've picked on one of the worse case scenarios from the [MLab data](https://docs.google.com/spreadsheets/d/1UTRucwLY2VC4944-cKunL9yUPJkjDKXDdIPk6GY3610/edit#gid=0). I'm emulating the connection the village of Colintraive has, which reportedly has a 450/230 Kbps connection speed. This is a connection speed equivalent to a 2G or slow 3G connection.
```

Comment:
I don't think we need to link to the data again, as it's already established at the beginning of the document:

```suggestion
Using WebPageTest we can roughly simulate the effect that packet loss has on performance for our users. For testing I picking a fairly content heavy page like the [past Prime Ministers](https://www.gov.uk/government/history/past-prime-ministers) page and a low spec Moto G4 Android device. For the connection speed I've picked on one of the worse case scenarios from the MLab data. I'm emulating the connection that the village of Colintraive has, which is reportedly a 450/230 Kbps connection speed, roughly equivalent to a 2G or slow 3G connection.
```